core/core.go

```diff
@@ -0,0 +1,372 @@
+package core
+
+import (
+	"io"
+	"math/big"
+
+	"github.com/spruce-solutions/go-quai/common"
+	"github.com/spruce-solutions/go-quai/consensus"
+	"github.com/spruce-solutions/go-quai/core/rawdb"
+	"github.com/spruce-solutions/go-quai/core/state"
+	"github.com/spruce-solutions/go-quai/core/state/snapshot"
+	"github.com/spruce-solutions/go-quai/core/types"
+	"github.com/spruce-solutions/go-quai/core/vm"
+	"github.com/spruce-solutions/go-quai/ethdb"
+	"github.com/spruce-solutions/go-quai/event"
+	"github.com/spruce-solutions/go-quai/params"
+	"github.com/spruce-solutions/go-quai/rlp"
+)
+
+type Core struct {
+	sl     *Slice
+	engine consensus.Engine
+}
+
+func NewCore(db ethdb.Database, chainConfig *params.ChainConfig, domClientUrl string, subClientUrls []string, engine consensus.Engine, vmConfig vm.Config) (*Core, error) {
+	slice, err := NewSlice(db, chainConfig, domClientUrl, subClientUrls, engine, vmConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Core{
+		sl:     slice,
+		engine: engine,
+	}, nil
+}
+
+// TODO
+func (c *Core) InsertChain(blocks types.Blocks) (int, error) {
+	for i, block := range blocks {
+		err := c.sl.Append(block)
+		if err != nil {
+			return i, err
+		}
+	}
+	return len(blocks), nil
+}
+
+func (c *Core) InsertHeaderChain(headers []*types.Header, checkFreq int) (int, error) {
+	return 0, nil
+}
+
+func (c *Core) InsertReceiptChain(blocks types.Blocks, receipts []types.Receipts, ancientLimit uint64) (int, error) {
+	return 0, nil
+}
+
+// InsertChainWithoutSealVerification works exactly the same
+// except for seal verification, seal verification is omitted
+func (c *Core) InsertChainWithoutSealVerification(block *types.Block) (int, error) {
+	return 0, nil
+}
+
+func (c *Core) SetTxLookupLimit(limit uint64) {
+}
+
+func (c *Core) Processor() *StateProcessor {
+	return c.sl.hc.bc.processor
```

Comment:
(nit) not sure if this is worth doing now, but we should be careful about what methods are exposed through the Core interface. We probably don't want block processing happening except through the Slice interface, no?